### PR TITLE
initial array functionality attempt for positionReader ws

### PR DESCRIPTION
**Pull request by ghjor**
Pr for ease of comm @Balugrizzly

### CODE REVIEW

**Comment by ghjor** [comment]
Not tested for player ws, can't seem to get the example in the README to work. Successful for other ws's

**Comment by Balugrizzly** [comment]
Can you elaborate on the first two sentences? More Info would be noice 
How do you know it does not work if you have not tested it?

File: controllers/ws/position/positionReader.go [hunk 1]
```diff
@@ -12,6 +12,10 @@ import (
 	"github.com/spf13/viper"
 )
 
+type Unmarshaler interface {
+	UnmarshalJSON([]byte) error
+}
+
```
Comment (Balugrizzly) [inline controllers/ws/position/positionReader.go:18]: What is this?

File: controllers/ws/position/positionReader.go [hunk 2]
```diff
@@ -29,55 +33,64 @@ func positionReader(c echo.Context, ws *websocket.Conn, ch chan error, timeoutCT
...
+				// TODO: Handle ID and production mode
 
-			c.Logger().Debug("Validating positionModel")
-			if !positionModel.IsValid() {
-				c.Logger().Debug("positionModel is NOT valid returning")
-				// NOTE: no Chan Timeout used
-				ch <- errors.New("positionModel Validation failed")
-				return
-			}
+        c.Logger().Debugf("positionModel: %+v", positionModel)
```
Comment (Balugrizzly) [inline controllers/ws/position/positionReader.go:66]: Are you logging a empty Struct?

File: controllers/ws/position/positionReader.go [hunk 3]
```diff
@@ -29,55 +33,64 @@ func positionReader(c echo.Context, ws *websocket.Conn, ch chan error, timeoutCT
...
+
+				if viper.GetBool("DEBUG") {
+					// Accept client provided ID in DEBUG mode
+					positionModel.ID = reqPosition.ID
+				}
+        
+        if reqPosition.ID <= 0 {
+          ch <- errors.New("missing/invalid ID")
+          return
+        }
```
Comment (Balugrizzly) [inline controllers/ws/position/positionReader.go:77]: Fmting?

File: controllers/ws/state/stateReader.go
Comment (Balugrizzly) [inline controllers/ws/state/stateReader.go:1]: This was unimplemented sortoff 
and is crossing paths with @Sarahtonein 
whats the status here?

**Review by Balugrizzly — CHANGES_REQUESTED** [review]
Comments from the positionreader file apply to the related other files

**Comment by ghjor** [comment]
> Can you elaborate on the first two sentences? More Info would be noice How do you know it does not work if you have not tested it?

Basically I couldn't get the player ws functional even without my array implementation. Sending data to the other ws's worked when I copied and pasted the readme example but I couldn't get any response when I sent to player, it just hangs. So I'm considering my array implementation untested for player ws. Somewhat semantics I think but definitely could have been clearer.

File: controllers/ws/position/positionReader.go (hunk 1 above)
Comment (ghjor) [reply controllers/ws/position/positionReader.go:18]: Something I forgot to remove from when I was trying out variable JSON structures implementation, my bad.

File: controllers/ws/position/positionReader.go (hunk 2 above)
Comment (ghjor) [reply controllers/ws/position/positionReader.go:66]: Removed

File: controllers/ws/position/positionReader.go (hunk 3 above)
Comment (ghjor) [reply controllers/ws/position/positionReader.go:77]: Yeah not sure what happened there. Usually I have autoformat on save

**Comment by Balugrizzly** [comment]
Lets go over this in vc hit me up when you are around 

> > Can you elaborate on the first two sentences? More Info would be noice How do you know it does not work if you have not tested it?
> 
> Basically I couldn't get the player ws functional even without my array implementation. Sending data to the other ws's worked when I copied and pasted the readme example but I couldn't get any response when I sent to player, it just hangs. So I'm considering my array implementation untested for player ws. Somewhat semantics I think but definitely could have been clearer.

File: controllers/ws/state/stateReader.go
Comment (Balugrizzly) [reply controllers/ws/state/stateReader.go:1]: What happened here?

Comment (Balugrizzly) [reply controllers/ws/state/stateReader.go:1]: > What happened here?

Why did you change this and what did you change?

**Comment by Balugrizzly** [comment]
As for the stateReader 
Why did you change this and what did you change? 
Talking about the change after the comment on it

File: controllers/ws/state/stateReader.go
Comment (ghjor) [reply controllers/ws/state/stateReader.go:1]: Reverted stateReader to its original state. Will save array implementation for when it's functional

**Comment by Balugrizzly** [comment]
Does this make it ready for merge in your eyes? @ghjor

**Comment by Balugrizzly** [comment]
Last commit looks good

**Pull request merged**